packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/remote_constraint_provider.dart

```diff
@@ -22,20 +31,128 @@ abstract class RemoteLoggingConstraintProvider {
 /// [LoggingConstraint] from an http endpoint periodically.
 /// {@endtemplate}
 class DefaultRemoteLoggingConstraintProvider
+    with AWSDebuggable, AWSLoggerMixin
     implements RemoteLoggingConstraintProvider {
   /// {@macro aws_logging_cloudwatch.default_remote_logging_constraint_provider}
   DefaultRemoteLoggingConstraintProvider({
     required DefaultRemoteConfiguration config,
     required AWSCredentialsProvider credentialsProvider,
   })  : _config = config,
-        _credentialsProvider = credentialsProvider;
+        _credentialsProvider = credentialsProvider {
+    _fetchAndCacheConstraintFromEndpoint(); // Fetch once immediately on initialize
+    _refreshConstraintPeriodically();
+  }
```

Comment:
Move these initializers to an `Future<LoggingConstraint?> init() async` method so that it can be awaited during initialization of the CloudWatchLoggerPlugin